### PR TITLE
Modified driver_radiation_lw to run on CPUs during an OpenACC run.

### PR DESCRIPTION
**Pull request by Pranay-Reddy-Kommera**
This is the 5th PR submitted in a pool of approximately 12 PRs.
The code changes include appropriate role checks and data transfers to/from CPU and GPU. The code changes allow the 'driver_radiation_lw' subroutine to execute on the CPU while any other dynamics/physics schemes are executing on the GPU. These temporary changes are required to keep track of any intermediate bugs originating as part of the Physics porting efforts. As the physics porting efforts progress, these changes will be removed if the data transfers become redundant.

Code Execution Status:

1. The code compiles and executes successfully on CPU.
2. Code compiles successfully with Dynamics on GPU and Physics on CPU. Code is not executed as the arrays/variables are not yet updated appropriately (will result in NaNs or incorrect answers) in the Physics. The code should be working by the end of all the PRs.

### CODE REVIEW

**Review by cenamiller — APPROVED** [review]

**Comment by cponder** [comment]
Code-changes look good from my end.
Also I ran the `7.x/benchmark/30km` case with these PRs
```893 895 896 902 903 904 905 906 907 908 909 910 911```
and didn't see any failure issue.

**Review by cponder — APPROVED** [review]

**Review by raghuraj19 — APPROVED** [review]

File: src/core_atmosphere/physics/mpas_atmphys_driver_radiation_lw.F [hunk 1]
```diff
@@ -402,8 +402,14 @@ subroutine radiation_lw_from_MPAS(xtime_s,configs,mesh,state,time_lev,diag_physi
...
- if (mpas_cpl%role_is(ROLE_INTEGRATE) .or. &
-     mpas_cpl%role_is(ROLE_RADIATION)) then
+ if (mpas_cpl%role_includes(ROLE_INTEGRATE)) then
+    !$acc update host(latCell, lonCell, aerosols, skintemp, snow, xice, &
+    !$acc xland, pin, ozmixm, sfc_albedo, sfc_emiss, cldfrac, o3clim, &
+    !$acc o3vmr, m_hybi, m_ps)
+ end if
+
+ if (mpas_cpl%role_includes(ROLE_INTEGRATE) .or. &
+     mpas_cpl%role_includes(ROLE_RADIATION)) then
```
Comment (mgduda) [inline src/core_atmosphere/physics/mpas_atmphys_driver_radiation_lw.F:412]: The original logic here was:
```fortran
 if (mpas_cpl%role_is(ROLE_INTEGRATE) .or. &
     mpas_cpl%role_is(ROLE_RADIATION)) then
```
which ensured that fields were transferred only if MPI tasks were performing an exclusive role (either INTEGRATE or RADIATION). The new logic,
```fortran
 if (mpas_cpl%role_includes(ROLE_INTEGRATE) .or. &
     mpas_cpl%role_includes(ROLE_RADIATION)) then
```
will transfer fields even in cases where MPI tasks are performing both INTEGRATE and RADIATION roles. Is this the intent?

File: src/core_atmosphere/physics/mpas_atmphys_driver_radiation_lw.F [hunk 2]
```diff
@@ -426,6 +432,12 @@ subroutine radiation_lw_from_MPAS(xtime_s,configs,mesh,state,time_lev,diag_physi
 
  end if
 
+ if (mpas_cpl%role_is(ROLE_INTEGRATE)) then
```
Comment (mgduda) [inline src/core_atmosphere/physics/mpas_atmphys_driver_radiation_lw.F:435]: Here we're using `role_is`, while what looks like the symmetric transfer on L.405 uses `role_includes`. Should the if-tests in both cases be the same?

**Comment by Pranay-Reddy-Kommera** [comment]
I have reverted 'role_includes' to 'role_is'. Please review the code.

**Review by mgduda — APPROVED** [review]

**Pull request merged**